content/en/blog/_posts/2024-08-14-Kubernetes-v1-31-Release.md

```diff
@@ -0,0 +1,299 @@
+
+With Kubernetes v1.31, classic DRA has a separate feature gate and needs to be enabled with `DRAControlPlaneController`. With such a control plane controller, a DRA driver can implement allocation policies that are not supported yet through structured parameters.
+
+This work was done as part of [KEP #3063](https://github.com/kubernetes/enhancements/issues/3063) by [SIG Node](https://github.com/kubernetes/community/tree/master/sig-node).
+    
+### Support for volumes based on OCI artifacts
+
+The Kubernetes community is moving towards fulfilling more Artificial Intelligence (AI) and Machine Learning (ML) use cases in the future. 
+
+One of the requirements to support these use cases is to support Open Container Initiative (OCI) compatible images and artifacts (referred as OCI objects) directly as a native volume source. This allows users to focus on OCI standards as well as enables them to store and distribute any content using OCI registries.
+
+Given that, the Image Volume Source feature is introduced as a new alpha feature in v1.31. This feature allows users to specify an image reference as volume in a pod while reusing it as volume mount within containers.
+
+This work was done as part of [KEP #4639](https://github.com/kubernetes/enhancements/issues/4639) by [SIG Node](https://github.com/kubernetes/community/tree/master/sig-node) and [SIG Storage](https://github.com/kubernetes/community/tree/master/sig-storage).
+
+### Exposing device health information through Pod status
+
+Expose device health information through the Pod Status is added as a new alpha feature in v1.31, disabled by default.
+
+Before Kubernetes v1.31, the way to know whether or not a Pod is associated with the failed device is to use the [PodResources API](/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/#monitoring-device-plugin-resources). 
+
+By enabling this feature, the field `allocatedResourcesStatus` will be added to each container status, within the `.status` for each Pod. The `allocatedResourcesStatus` field reports health information for each device assigned to the container.
+ 
+
+This work was done as part of [KEP #4680](https://github.com/kubernetes/enhancements/issues/4680) by [SIG Node](https://github.com/kubernetes/community/tree/master/sig-node).
+
+
+### Finer-grained authorization based on selectors
+
+This feature allows webhook authorizers and future (but not currently designed) in-tree authorizers to allow list and watch requests based on field and label selectors.  It is now possible for an authorizer to express: this user cannot list all pods, but can list all pods where `.spec.nodeName=foo`.  Combined with CRD field selectors (moving to beta in 1.31), it is possible to write more secure per-node extensions.
```

Comment:
(nit)
I like to avoid pseudocode in articles as it can encourage readers to build a misleading mental model of
how Kubernetes works.

Do we mention CRD field selectors in the article? If not, I think we should.
```suggestion
This feature allows webhook authorizers and future (but not currently designed) in-tree authorizers to
allow **list** and **watch** requests, provided those requests use label and/or field selectors.
For example, it is now possible for an authorizer to express: this user cannot list all pods, but can list all pods where `.spec.nodeName` matches some specific value. Or to allow a user to watch all Secrets in a namespace
that are _not_ labelled as `confidential: true`.
Combined with CRD field selectors (also moving to beta in 1.31), it is possible to write more secure
per-node extensions.
```